[Feb 4 merge] Review GitHub lessons 0-3

lwasser opened the pull request:
These lessons are already online to meet a deadline. However, I wanted to make sure we could have a normal review process on them!

Therefore, this is a hacked branch that allowed me to create a pr with the 4 files in the lessons that can be reviewed as "new" files. 

If you want to see the lessons rendered, you can either check Ci OR you can have a look [here for context](https://www.pyopensci.org/lessons/github-git/0-first-contribution.html#). these lessons do have some images and we can always add more. 

I'm looking for feedback on how the lessons read, The goal is to use them in our sprints and to make it easier for people to make first contributions to open source (which we normally have happen in our sprint events!).

lwasser commented on github-git/3-fork-repo.md:
```diff
@@ -0,0 +1,106 @@
+**1. Fork the pyOpenSci practice GitHub repository**
+*******
+
+Fork the <a href="http://www.github.com/pyopensci/repo-here" target="_blank">pyOpenSci demo repository on GitHub</a>. Remember that a fork is a copy of a repository that is owned by someone else or an organization that lives in your GitHub account.
```
```suggestion
Fork the [pyOpenSci example repo](https://github.com/pyOpenSci/pyos-demo-package-contribute). Remember that a fork is a copy of a repository that is owned by someone else or an organization that lives in your GitHub account.
```

lwasser commented on github-git/3-fork-repo.md:
```diff
@@ -0,0 +1,106 @@
+**2. Open the file that you proposed changes to in the issue you selected in the [how to identify an issue lesson](2-identify-issue).**
+*******
+
+In the previous issue, you selected or identified something that you wanted to fix in a GitHub repo. Ideally, if this is your first contribution, that fix is small and could potentially be completed in the GitHub online interface. 
+
+* Select the file that you decided to work on in GitHub
+* Click on the edit <kbd><i class="fa-solid fa-pencil"></i></kbd> button in the GitHub interface.
+* Make the edits that you proposed to make.
+* Hit the <kbd>commit</kbd> button to save your edits
```
```suggestion
```
The next lesson is about editing and commiting; this information doesn't belong here.

jsdodge commented on github-git/1-get-to-know-repo.md:
```diff
@@ -0,0 +1,78 @@
+### Answer these questions
+
+* Does the repository accept contributions? If so, what types of contributions are accepted?
+* Does the repository use specific code or text formatting standards or liters?
```
I think there is a typo: "liters" --> "linters"

lwasser replied:
good catch!!

jsdodge commented on github-git/2-identify-issue.md:
```diff
@@ -0,0 +1,121 @@
+
+In this lesson you will create or comment on an issue that you'd like to fix in GitHub repository that you don't own. 
+
+:::{admonition} Activity: Create (or find) an issue for a bug that you'd like to fix 
```
Is there a reason you don't just direct the user to open an issue to fix a specific error in the code or documentation? If the purpose is mainly to walk people through the process of opening an issue, it seems unnecessary to ask the user to look for something to fix in a dummy repository. This confused me initially.

lwasser replied:
ahhh - such a good point. @jsdodge we're trying to balance the reality that an issue might already exist vs the activity here which will be hands on.

Should i clarify that the activity is just to create one but in the real world, you should check that one exists first perhaps?

jsdodge replied:
Yes, I think that would be clearer.

lwasser replied:
Perfect!!

jsdodge reviewed (commented):
Hi @lwasser , I added a few inline comments to highlight a few minor issues. I stopped before reviewing `3-fork-repo.md`. This is all very helpful for a newcomer like me!

lwasser commented:
@all-contributors please add @jsdodge for review

ucodery commented on github-git/2-identify-issue.md:
```diff
@@ -0,0 +1,121 @@
+   * Explain the problem or fix you’re proposing.
+   * For bugs, include steps to reproduce the issue.
+   * For documentation, link to the page and describe what you’d like to update or enhance.
+
```
Another good step is to
for bugs:
- describe what you expected would happen, and what actually happened

for new features:
- describe what will happen once it is implemented

lwasser replied:
Added!!

ucodery commented on github-git/2-identify-issue.md:
```diff
@@ -0,0 +1,121 @@
+Examples of less useful issue titles:
+* Update documentation.
+* Fix formatting.
+* Be Constructive: Avoid blaming or harsh criticism. Instead, frame concerns as suggestions.
```
```suggestion

Be Constructive: Avoid blaming or harsh criticism. Instead, frame concerns as suggestions.
```

ucodery commented on github-git/2-identify-issue.md:
```diff
@@ -0,0 +1,121 @@
+
+* [Fork the repository](fork-repo) if you haven't already.
+* [Make your changes on a new branch in your fork.](4-edit-commit-files)
+* [Submit a pull request](5-pull-request) with your updates. 
```
This might be something you plan to cover in pull-request, but the Issue and PR should be linked together somehow, whether that is in the PR title or a PR comment mentioning the issue, or an issue comment mentioning the PR.
And once both are open, this guide should take care to be clear about which thread recommended communications should go in.

lwasser replied:
Good idea. I think it would be great to reinforce this a few times. so I'll incorporate it up a bit higher just to raise awareness early on!!

lwasser replied:
ok added - i'll remember to mention this in the pr lesson too!!

ucodery commented on github-git/3-fork-repo.md:
```diff
@@ -0,0 +1,100 @@
+**1. Fork the pyOpenSci practice GitHub repository**
+*******
+
+Fork the [pyOpenSci example repo](https://github.com/pyOpenSci/pyos-demo-package-contribute). Remember that a fork is a copy of a repository that is owned by someone else or an organization that lives in your GitHub account.
```
```suggestion
Fork the [pyOpenSci example repo](https://github.com/pyOpenSci/pyos-demo-package-contribute). Remember that a fork is a copy of a repository that is owned by someone else.
```

A mild suggestion, but I don't think a reader wanting to learn the basics of github is going to manage multiple namespaces under one GitHub account and trade PRs between them. This seem to distract from the (slightly simple) definition.

sneakers-the-rat replied:
in training my lab on git (most are electrical engineers not software ppl), another point of confusion is "what's the difference between a fork and a branch," not sure if that's worth expanding on here bc not sure what level we're pitching this lesson at

ucodery commented on github-git/3-fork-repo.md:
```diff
@@ -0,0 +1,100 @@
+
+When a repository is stored on **GitHub.com**, it is assigned a unique URL (i.e. link on the **GitHub.com** website) that can be used to find the repository and access its files. While repositories on **GitHub.com** can be made either public or private, the default is public for free **GitHub** accounts.
+
+In either case (public or private), the URL links to a **GitHub** repository always follows the same format: 
```
```suggestion
In either case (public or private), the URL links to a **GitHub** repository always follow the same format: 
```

ucodery commented on github-git/3-fork-repo.md:
```diff
@@ -0,0 +1,100 @@
+
+`https://github.com/username/repository-name`
+
+The username is the username owner of the repo owner. The username can either be an individual such as `lwasser` (your **GitHub** username), or it can represent an organization such as `pyOpenSci`.
```
```suggestion
The username is the owner of the repository. The username can either be an individual such as `lwasser` (your **GitHub** username), or it can represent an organization such as `pyOpenSci`.
```

ucodery commented on github-git/3-fork-repo.md:
```diff
@@ -0,0 +1,100 @@
+
+The username is the username owner of the repo owner. The username can either be an individual such as `lwasser` (your **GitHub** username), or it can represent an organization such as `pyOpenSci`.
+
+For example, the repository that you will work within this lesson is owned by `pyopensci.` Therefore, the URL looks like this:
```
```suggestion
For example, the repository that you will work within this lesson is owned by `pyopensci`. Therefore, the URL looks like this:
```

ucodery commented on github-git/3-fork-repo.md:
```diff
@@ -0,0 +1,100 @@
+
+It also means you can update your fork anytime, as the parent repository is updated to ensure it stays in sync.
+
+2. If you have already forked the repository but some time has passed. You should consider updating or syncing your fork. GitHub has a sync button that you can use to do this (`pyopensci/repo-name`). This will ensure that all of the files in your repository are current.
```
```suggestion
2. If you have already forked the repository but some time has passed. You should consider updating or syncing your fork. GitHub has a sync button that you can use to do this (<kbd>Sync fork</kbd>). This will ensure that all of the files in your repository are current.
```

sneakers-the-rat replied:
omg first time seeing the `<kbd>` element

<kbd>Comtrol</kbd> + <kbd>Elt</kbd> + <kbd>Balete</kbd>

ucodery commented on github-git/3-fork-repo.md:
```diff
@@ -0,0 +1,100 @@
+
+Fork the [pyOpenSci example repo](https://github.com/pyOpenSci/pyos-demo-package-contribute). Remember that a fork is a copy of a repository that is owned by someone else or an organization that lives in your GitHub account.
+
+**2. Open the file that you proposed changes to in the issue you selected in the [how to identify an issue lesson](2-identify-issue).**
```
This point seems out of context of the rest of the page, and more like the starting point of edit-commit-files.

ucodery commented on github-git/3-fork-repo.md:
```diff
@@ -0,0 +1,100 @@
+---
+
+(fork-repository)=
+# How to Fork a GitHub Repository 
```
This is a different workflow, and so might not be easy to teach along side the lesson you have now, but if all edits happen in the browser, navigating to the file to be edited *first* and clicking the edit pen button will take the user to an alternative `Fork this repository` page (only if you haven't yet forked).

lwasser replied:
OH gosh - so true. I discovered this when creating the screen captures! 

What is so hard is that there are actually multiple potential paths here! I thought about an overview page (i have a [draft online here](https://www.pyopensci.org/lessons/github-git/ways-to-contribute.html#start-here-if-you-haven-t-yet-forked-the-repository) that also shows them the various paths in case they happen to do what you just mentioned! 

But we stick with "first principles," so to speak, in this lesson, which would be the traditional issue, fork, work, commit, pr route. 

I want to show them one way clearly. But then maybe have a resource page that shows alternative paths? 

what do you think about that?

ucodery replied:
Probably best to stick to one workflow. And what you have is the more general, probably seen as more "proper" as it is how you would make a fix locally.

Best to keep it simple. I'll just note that every PyOS sprint I have gone to, I've encouraged attendees to go straight for the edit button.

lwasser replied:
Yup. ok cool. yes i think we can in the future show them those workflows as well (i've done that too in sprints). 
i've started this page below:

https://www.pyopensci.org/lessons/github-git/ways-to-contribute.html

to capture the various ways you can create a pr! That way we have that documented for sprints too (we can work on it together via a future review as well). i plan to open more pr's here with a fresh version of each lesson so we can review it as if it's new after i get this grant deliverable in today (its done!).

ucodery commented:
> If you want to see the lessons rendered, you can either check Ci OR you can have a look [here for context](https://www.pyopensci.org/lessons/github-git/0-first-contribution.html#). these lessons do have some images and we can always add more.

Also, for me, <https://www.pyopensci.org/lessons/github-git/0-first-contribution.html> is returning 404

sneakers-the-rat commented on github-git/0-first-contribution.md:
```diff
@@ -0,0 +1,66 @@
+## Documentation counts! 
+
+It's common for your first contribution
+to be to documentation! Documentation contributions can be incredibly valuable as developers often don't have time to work on documentation and code!
```
"...incredibly valuable, as your perspective is the best way for developers who are too familiar with the project to know what is unclear to everyone else."

or something like that, centering the learner <3

sneakers-the-rat commented on github-git/0-first-contribution.md:
```diff
@@ -0,0 +1,66 @@
+
+Contributing to open source in a public space like GitHub can feel intimidating. You may not know the project maintainers, feel unsure about your GitHub skills, or wonder where to begin.
+
+This lesson series will teach you how to collaborate on code and documentation using GitHub. It will help you confidently contribute to a GitHub repository and build skills to collaborate with colleagues (some of whom you may not have met in real life!). 
```
what's the context for these lessons? do the students already know rough contours of what open source is, why one might want to contribute to something, etc.?

lwasser replied:
I suspect we need a context page. I"m going to update this page (i just merged Jeremiahs inline stuff). I reorganized this (but I want your feedback) to be a set of activities with links to the background for each. i'm not sure if this is the best way or not.

but this is what i imagine

we have sprints regularly and people come that are new to contributing. I can send them to this page and it walks them through all of the steps with background information on each following page on how to implement each. I've never designed a lesson like this before, so I'm not sure how it will land. So i think we might want to add a context page first about contributing and github (potentially?) That context might be something like [this page](https://www.pyopensci.org/lessons/github-git/intro.html) with more focus on open source.  

these pages are online because i have a grant deadline but i still want to do reviews like we always do ❤️

sneakers-the-rat commented on github-git/1-get-to-know-repo.md:
```diff
@@ -0,0 +1,78 @@
+---
+
+(get-to-know-repo)=
+# Get to Know a (new to you) GitHub Repository
```
in my experience, one of the most intimidating things from ppl i have helped mentor who are venturing out and doing their first contribs is fear of reprisal/actually getting yelled at/PRs getting closed without discussion/etc. I think *somewhere* in this lesson it's worth spending time introducing the idea that each project has its own culture, history, and norms, and that navigating those norms is as important as knowing the technical side of how to make a contrib. I think this is already present in the lesson in different places, but I think here at the intro to this section would be good place for that framing to be made explicit.

Part of that that I think we in particular can do well where other tutorials like this fail is in being reassuring - open source can be wonderful, but you will also inevitably accidentally wander into a pit of assholes, and you should know that as long as you are doing your best to understand the norms and be respectful that you aren't doing anything wrong. This is to put in perspective *why* there is so much ritual around making a contrib - e.g. high volume projects they are dealing with a huge amount of work all the time, so they have to make rules and processes to make that manageable. I think without that context it might be a little unclear why you have to go through all this rigamarole when the "open PR" button is right there and i can click it at any time

lwasser replied:
awwww such a good point Jonny!! 

In some of the other lessons i started adding little social cue tips but they are all a bit too maintainer focused (avoid getting yelled at) and could be more supportive. 

My one question is - should the overview / culture stuff be in its own intro or here ? it could fit here well too especially when talking about code of conduct, etc.

lwasser replied:
I did some redesigning here, sprinkling social cues throughout. But I welcome other places for more encouraging language. I don't think i've hit this comment well yet. But it's the most important one. I think the context you mention is really important. 

When i've worked with people it's always pyOpenSci that they are contributing too. they are always afraid of breaking something. 

your point is what I've personally experienced, especially early on - the fear of doing things wrong and potential repercussions. how do we gently nudge and support here?!

sneakers-the-rat commented on github-git/0-first-contribution.md:
```diff
@@ -0,0 +1,66 @@
+The steps for making your first contribution to an open source repo. In this lesson, you will do all of your work on GitHub.com, meaning you don't need to clone or make a copy of your repository locally. This is a great way to get started with contributing. It minimizes the command line skills that you will need to use git. 
+:::
+
+## Documentation counts! 
```
This comes a little bit out of nowhere, we are still in overview mode and we're already talking about what kind of PRs we can make. Seems like this would fit better in "identify issue"

lwasser replied:
It does! I'm going to address these comments now and then update this doc, as I did rework this section a bit. But now i see Jeremiah's comment below about small pr's not being welcome in some projects! i just didn't know that was a thing.

sneakers-the-rat commented on github-git/2-identify-issue.md:
```diff
@@ -0,0 +1,121 @@
+
+In this lesson you will create or comment on an issue that you'd like to fix in GitHub repository that you don't own. 
+
+:::{admonition} Activity: Create (or find) an issue for a bug that you'd like to fix 
```
Along the same lines as @jsdodge 's comment on this line, this comes a little bit out of order for me. So in the flow of the lesson we have just finished getting to know the repository. Presumably in reality we are starting from the place of already using the package and noticing something wrong as step -1, and while I understand the idea behind the dummy repository, it seems odd to start this section with "go look for a bug." maybe this should come at the end of this document?

sneakers-the-rat commented on github-git/2-identify-issue.md:
```diff
@@ -0,0 +1,121 @@
+Once you have submitted an issue and someone has responded positively, you can begin working on the changes in your fork. [You will learn how to fork a repo, next.](fork-repo)
+:::
+
+As you scan the issues, look for beginner-friendly labels. Beginner-friendly labels are a clear message that the project welcomes new contributors. Many projects label issues as **good first issue** or **beginner-friendly** to highlight approachable tasks that are approachable for new contributors.
```
maybe something along the lines of "picking up existing issues, even though they might not be related to the current problem you are experiencing, are a good way of getting to know and building rapport with the developers. remember you're trying to propose changes to something potentially very dear to them! a small bugfix for a known issue to signalt hat you can come in peace can go a long way"

sneakers-the-rat commented on github-git/2-identify-issue.md:
```diff
@@ -0,0 +1,121 @@
+
+## Open or comment on an issue
+  
+If you **have an idea already of what you'd like to work on**, check if an issue exists for it. If it does, comment on the issue to indicate your interest. If not, create a new issue and wait for feedback.
```
ya ya ya.

I think that we should split into two sections "search open issues," "raise an issue," and then expand a bit on searching open issues, stuff like:
- how to search: search using keywords, but also the code symbols/variable names/etc. that are involved with the problem
- searching both open and closed issues: is this a known bug, or in the case of closed issues, have the maintainers decided this is a `wontfix`?
- as @ucodery says, assess the status of the work - is there a plan in place? is someone already working on it? 
- follow backlinks to related issues if you find something that is a near match but not exactly what you had in mind,

etc. 

along with some context like "you want to demonstrate that you are willing to put at least as much work into raising the issue as you expect the maintainers to expend responding to it, and demonstrating that you have already searched is a good way to show goodwill"

lwasser replied:
This is all so good. ill work on it tomorrow! thank you everyone. I really wanted to add all of these (what i'm calling social cues) to these lessons. the stuff i NEVER see in contribute to open source tutorials but that is actually the hardest stuff :) so thank you for raising all of these items.

sneakers-the-rat commented on github-git/2-identify-issue.md:
```diff
@@ -0,0 +1,121 @@
+1. Click on the new issues button. If there is a template that you can select for the issue, use one.
+
+1. Write a **clear title** summarizing what you’d like to fix. Some repositories use templates—fill them out if available.
+1. Be **specific about what you'd like to fix**:
```
+ "an issue should ideally be about a single thing" (even though what a "single thing" is is an ill-defined problem)

sneakers-the-rat commented on github-git/2-identify-issue.md:
```diff
@@ -0,0 +1,121 @@
+
+1. Click on the new issues button. If there is a template that you can select for the issue, use one.
+
+1. Write a **clear title** summarizing what you’d like to fix. Some repositories use templates—fill them out if available.
```
some examples would probably be good here. 

bugs:
* bad: "program doesn't work"
* bad: "{copying a whole raw traceback}"
* good: "{error} when {condition}" - "OSError when passing relative path to `some_function`"

feature requests:
* bad: "Some suggestions" - too vague
* bad: "Improving the user experience by adding an extra parameter "relative=True/False" to `some_function` to allow the user to specify whether a path is relative or absolute" - too chatty
* good: "Add `relative` parameter to `some_function`"

sneakers-the-rat replied:
Aha, i see this is down below on line 91 and company. Leaving this here just to say "reading this as a newbie, i am a little confused at what is meant by a clear title, and that might be a bit distracting"

lwasser replied:
Good catch! i moved it up to here! And put it in an admonition with a table :) 
<img width="861" alt="Screenshot 2025-01-30 at 6 25 25 PM" src="https://github.com/user-attachments/assets/9826b78f-3bbb-4927-8295-9d4d26c53d12" />

sneakers-the-rat commented on github-git/2-identify-issue.md:
```diff
@@ -0,0 +1,121 @@
+> Hi team 👋, just following up on my issue. Let me know if you’d like me to move forward with a pull request or if there’s anything I should address first.
+
+
+## Tips for submitting good issues
```
I think that a bit of restructuring of this section of the lesson might help this feel more situated in the flow of the lesson, currently it feels a little bit like a "grab bag" of (good) advice and we end up either repeating ourselves or missing information elsewhere (e.g. re: "good title" comment above).

currently:
```
## Open or comment on an issue
### How to create a new issue
## Tips for submitting a good issue
## What happens next
```

could be something like

```
## Search for existing issues
## Continuing an open issue
## Opening a new issue
### Issue templates (they'll probably experience this first if they're contributing to an established project)
### Components of a good issue
## What happens next
```

where we split out each of the stages here into their own sections, and then within "components of a good issue" we can have subsections for each of the parts of an issue that can be a bit more like a "formula" for newbies

- titles
- context/motivation
- cause of error/MRE
- result/error message
- expectation, "what would close this issue"
- links to related issues

I think these are all present in the document, just a little out of order, and it might be just a me thing but having these split across heading sections where part of this is introduced in heading 1.1 and another part in 2 makes it harder for me to see what things are related and which aren't.

lwasser replied:
Yes this is much better! i'll push a version of this tomorrow 🚀  (i don't have stuff on templates yet).

sneakers-the-rat commented on github-git/3-fork-repo.md:
```diff
@@ -0,0 +1,100 @@
+`https://github.com/pyopensci/repo-name`
+
+(fork-repo)=
+## What is forking a GitHub repository?
```
similar comment re: structure as in section 2 of the lesson - two big ideas in this section: "what is a fork" and "how to make a fork"

currently

```
## How to fork a GitHub repository
## Who owns a GitHub repository?
## What is forking a GitHub repository?
## A few tips about forking a repo
```

could be

```
## What is a fork?
### ... info about what forks are
## How to fork a GitHub repository
### ... info on how to make a fork
```

lwasser replied:
i will reorg some tomorrow. One thing I was considering here was put the action items at the top, considering people have limited attention spans these days. And then explanation below. 

I can't decide. We are living in a world of video shorts and such so i wondered what would happen if the "how to" was at the top and the "understand what you just did" came later. As an educator, it's not my preference, but do people online have the attention span to read it all, or do they want to skim and go back to learn?

thoughts?

sneakers-the-rat commented on github-git/3-fork-repo.md:
```diff
@@ -0,0 +1,100 @@
+
+It also means you can update your fork anytime, as the parent repository is updated to ensure it stays in sync.
+
+2. If you have already forked the repository but some time has passed. You should consider updating or syncing your fork. GitHub has a sync button that you can use to do this (`pyopensci/repo-name`). This will ensure that all of the files in your repository are current.
```
re: this, i think it would be worth having a "forks v. branches" subsection in "what is a fork?" section - along the lines of "a fork is an independent copy of a repository, you have a double of all the branches (or maybe just `main` depending on options), in your fork, so if they do work upstream, your version of `main` will get out of date." which would give this a place to live, as well as allow us to discuss stuff like why you should work on branches even in your fork so you can sync `main` when needed and leave links for further reading on rebasing/merging to keep a fork up to date.

lwasser replied:
Yes definitely a good call. I started a graphic that captures this but will add a bit more text either today or next week. We can leave this open for longer if needed to really get the lessons right now that the grant deliverable is in. 
<img width="1277" alt="Screenshot 2025-01-31 at 12 59 36 PM" src="https://github.com/user-attachments/assets/e03f876d-72d0-41c9-8633-81a6dee791ee" />

the last deliverable is community review 🚀 and it makes everything better.

sneakers-the-rat commented:
frick i thought i was gonna make just one or two comments but then i got to reading and made a billion. sorry for not collecting in a review and also looking very good and this is yet another lesson that i badly need a good version of all the time! thank you for writing this!

lwasser commented on github-git/0-first-contribution.md:
```diff
@@ -0,0 +1,66 @@
+It's common for your first contribution
+to be to documentation! Documentation contributions can be incredibly valuable as developers often don't have time to work on documentation and code!
+
+Fixing typos and other small documentation issues makes for great, easy-to-create (and review) first pull requests. In fact, if you find that the contributing information isn't detailed enough, that could also be your first contribution!
```
Oh wow - @ucodery i didn't know that!! some won't accept those types of pr's at all? All of my first contributions were those kinds. can you help me better understand this so we can capture it?

lwasser replied:
Oh that's interesting. Given that, maybe we need a short blurb that better explains that issue. i'll turn this into an issue in this repo to ensure we acknowledge it. I could see drive by contributions when someone is just trying to get github cred being really challenging

lwasser commented on github-git/0-first-contribution.md:
```diff
@@ -0,0 +1,188 @@
+    "og:image:alt": An image that shows the steps for contributing to open source on GitHub.
+---
+
+# Mastering GitHub Collaboration Skills 
```
@sneakers-the-rat, have a look at this version and see what you think. it's a different approach - but it gives the learner the steps first and then it sends them to pages with more info. (if they need it). i just don't know how this will land with people.

lwasser commented on github-git/0-first-contribution.md:
```diff
@@ -0,0 +1,188 @@
+
+********
+
+## Many people's first contributions are to documentation 
```
from @sneakers-the-rat try to incorporate the learner's perspective here

* Sometimes, a learner will notice documentation bugs that a maintainer misses.

lwasser commented on github-git/0-first-contribution.md:
```diff
@@ -0,0 +1,188 @@
+
+## Many people's first contributions are to documentation 
+
+In this lesson, you make your first open source contribution by updating documentation (including docstrings) rather than code. Many people start with small fixes like typos, which are simpler to contribute as you don't always need a development environment but are still valuable. Developers often focus on code, leaving less time for writing and editing documentation. So your small contributions  ...
```
from @ucodery - some projects do NOT welcome small pr's. i need to know more about that to write about it! i had no idea that was a thing and I've never encountered it. Jeremiah, any feedback is welcome!!

lwasser commented on github-git/1-get-to-know-repo.md:
```diff
@@ -0,0 +1,78 @@
+### Answer these questions
+
+* Does the repository accept contributions? If so, what types of contributions are accepted?
+* Does the repository use specific code or text formatting standards or liters?
```
```suggestion
* Does the repository use specific code or text formatting standards or linters?
```

lwasser commented on github-git/0-first-contribution.md:
```diff
@@ -0,0 +1,188 @@
+
+## Many people's first contributions are to documentation 
+
+In this lesson, you make your first open source contribution by updating documentation (including docstrings) rather than code. Many people start with small fixes like typos, which are simpler to contribute as you don't always need a development environment but are still valuable. Developers often focus on code, leaving less time for writing and editing documentation. So your small contributions  ...
+
+Further, if you go to contribute and find that the contributing guidelines lack detail, that could also be a useful place for you to start. 
```
```suggestion
In this lesson, you make your first open source contribution by updating documentation (including docstrings) rather than code. Many people start with small fixes like typos, which are simpler to contribute as you don't always need a development environment but are still **highly valuable**. 

### <i class="fa-solid fa-rocket"></i> Fresh eyes make a difference! <i class="fa-solid fa-rocket"></i>

When you’re new to a project, you may notice gaps and unclear explanations that long-time contributors might overlook. Maintainers and developers are often too close to their own work to see what’s missing. Your perspective helps make the project more accessible to future contributors.

By catching confusing phrasing, outdated information, or missing details (even in contributing documentation), you improve the onboarding experience for others while making a meaningful first contribution!
```

lwasser replied:
@sneakers-the-rat this might capture that idea of fresh eyes are always better  emphasize what the new contributor offers...

lwasser commented:
Hi, y'all 👋🏻 
I've updated several pages, but for this one, I want to incorporate some of the new content in the issue page, so that one isn't updated at all yet.  i'll ping y'all when there is a new version up. 

I'm working in two places - locally to meet my Feb 1 grant deadline and here where I want all the input. We can always leave this open for a bit longer to really nail things down before merging. I have some graphics, too, that I'll try to share here and, if not here,  in the next pr. 

Thank you as always for being so collaborative in developing these lessons. your feedback and effort in making lessons better is one of my parts of lesson development here at pyOpenSci ✨

lwasser commented:
Ok y'all rock
[Here is where we are if you want to check out a rendered view. ](https://output.circle-artifacts.com/output/job/764be426-0aef-47cb-984f-bac07de66b8d/artifacts/0/html/github-git/0-first-contribution.html#) 

I've incorporated *I think* all of the comments and updated the lessons. In the preview you can see the graphics and such, too. Please ignore the later lessons; we will do a separate review for that. 

The plan now is:
* Leave this pr open for a few more days to see if we get more feedback (please leave comments on what I've changed and let me know if you think it meshes better
* Merge a separate pr version of these lessons (today) for the grant (but we can still do more work here!!)
* Next week, I'll open a new pr for the edit, commit, and pr lessons 🚀 which are well underway
* I'm also going to work on the context lesson, which will become another separate pr that I look forward to your feedback on.
thank you everyone!!

lwasser commented:
Graphics

fork repo

![fork-repo](https://github.com/user-attachments/assets/068fcb20-568d-4ece-8490-4e39a14e9372)

lwasser commented:
contributing path

![image](https://github.com/user-attachments/assets/dd608f27-1a33-48e3-8f6e-9214e9950bda)

lwasser commented:
> > If you want to see the lessons rendered, you can either check Ci OR you can have a look [here for context](https://www.pyopensci.org/lessons/github-git/0-first-contribution.html#). these lessons do have some images and we can always add more.
> 
> Also, for me, https://www.pyopensci.org/lessons/github-git/0-first-contribution.html is returning 404

Sorry - https://www.pyopensci.org/lessons/github-git/your-first-contribution.html I renamed the files removing the numbers from them.

lwasser commented:
> frick i thought i was gonna make just one or two comments but then i got to reading and made a billion. sorry for not collecting in a review and also looking very good and this is yet another lesson that i badly need a good version of all the time! thank you for writing this!

Don't worry about this @sneakers-the-rat for me, my github notifications are carefully curated so i saw all of the comments at once - no big deal at all. I'm so used to pr's with hundreds of comments it's not an issue for me when people forget to hit the review button anymore. :) thank YOU  (as always!!) for all of the great feedback. lessons are already monumentally better.

lwasser commented:
ok everyone. I'm going to merge this as planned today. I will be opening up additional lessons for review later this week. we can do a rereview if need be of lessons in the future as well.

lwasser commented:
merging this now!! lessons will be live in about 10 minutes 🚀
